src/diff/children.js

```diff
@@ -175,17 +177,6 @@ export function diffChildren(
 			}
 
 			if (childInternal._flags & COMPONENT_NODE) {
-				// @TODO: this needs to happen unconditionally in patch()
-				if (
-					childInternal != null &&
-					childVNode.props &&
-					childVNode.props.children != null &&
-					childVNode.props.children === childInternal.props.children
-				) {
-					startDom = reorderChildren(childInternal, startDom, parentDom);
-				} else {
-					startDom = nextDomSibling;
-				}
```

Comment:
This was an area where I don't really understand the implications of this change, but it just worked. At some point I'd like to dig into this more to understand what it was here originally to see if we are missing some edge cases or something....

Reply:
I'm good with the change if all the tests pass. If this piece of code was necessary we should be able to run into a case and turn that into a test. For now I'm good with removing it 👍